File: packages/google_mlkit_document_scanner/android/src/main/java/com/google_mlkit_document_scanner/GoogleMlKitDocumentScannerPlugin.java
package com.google_mlkit_document_scanner;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.MethodChannel;

public class GoogleMlKitDocumentScannerPlugin implements FlutterPlugin, ActivityAware {
    private static final String channelName = "google_mlkit_document_scanner";
    private MethodChannel channel;

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
        channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), channelName);
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        channel.setMethodCallHandler(null);
    }

    @Override
    public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
        channel.setMethodCallHandler(new DocumentScanner(binding));
    }

    @Override
    public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
        channel.setMethodCallHandler(new DocumentScanner(binding));
    }

    @Override
    public void onDetachedFromActivityForConfigChanges() {
    }

    @Override
    public void onDetachedFromActivity() {
    }
}

File: packages/google_mlkit_document_scanner/android/src/main/kotlin/com/google_mlkit_document_scanner/GoogleMlKitDocumentScannerPlugin.kt
package com.google_mlkit_document_scanner

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodChannel

class GoogleMlKitDocumentScannerPlugin :
    FlutterPlugin,
    ActivityAware {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, CHANNEL_NAME)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        channel.setMethodCallHandler(DocumentScanner(binding))
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        channel.setMethodCallHandler(DocumentScanner(binding))
    }

    override fun onDetachedFromActivityForConfigChanges() {}

    override fun onDetachedFromActivity() {}

    companion object {
        private const val CHANNEL_NAME = "google_mlkit_document_scanner"
    }
}
